jenkins/bootstrap.py

```diff
@@ -537,24 +536,34 @@ def build_name(started):
     return os.environ[BUILD_ENV]
 
 
-def setup_credentials():
-    """Run some sanity checks and activate gcloud."""
+def setup_credentials(robot, upload):
+    """Activate the service account unless robot is none."""
     # TODO(fejta): stop activating inside the image
     # TODO(fejta): allow use of existing gcloud auth
+    if robot:
+        os.environ[SERVICE_ACCOUNT_ENV] = robot
+    if not os.getenv(SERVICE_ACCOUNT_ENV) and upload:
+        logging.warning('Cannot --upload=%s, no active gcloud account.', upload)
+        raise ValueError('--upload requires --service-account')
+    if not os.getenv(SERVICE_ACCOUNT_ENV) and not upload:
+        logging.info('Will not upload results.')
+        return
     if not os.path.isfile(os.environ[SERVICE_ACCOUNT_ENV]):
```

Comment:
isn't this always  is going to always fail if `robot` is `None`? (unless users explicitly set `GOOGLE_APPLICATION_CREDENTIALS` in the environment... but then I don't understand why we'd need `--service-account`.)